### PR TITLE
Add support for loading images from swift backend

### PR DESCRIPTION
**Pull request by paladox**
Using config setting

### CODE REVIEW

**Comment by Universal-Omega** [comment]
Shouldn't we use FileBackendMultiWrite for the swift migration to write to both for now? The purpose of that states "useful for transitioning from one backend to another" which is exactly what we are doing here.

**Comment by paladox** [comment]
> Shouldn't we use FileBackendMultiWrite for the swift migration to write to both for now? The purpose of that states "useful for transitioning from one backend to another" which is exactly what we are doing here.

No, as we're migrating in stages. We aren't doing it in one big swoop.

**Review by Universal-Omega — CHANGES_REQUESTED** [review]
There is also some minor code quality things I could point out, but probably not 100% needed right now and pretty minor for the most part.

File: FileBackend.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	'scriptDirUrl' => $wgScriptPath,
+	'url' => $wgUploadBaseUrl ? $wgUploadBaseUrl . $wgUploadPath : $wgUploadPath,
+	'hashLevels' => $wgHashedUploadDirectory ? 2 : 0,
+	'thumbScriptUrl' => $wgThumbnailScriptPath,
+	'transformVia404' => !$wgGenerateThumbnailOnParse,
+	'deletedDir' => $wgDeletedDirectory,
+	'deletedHashLevels' => $wgHashedUploadDirectory ? 3 : 0,
+	'isPrivate' => $wmgPrivateUploads,
+	// new folders need to be added to puppet/modules/swift/files/SwiftMedia/miraheze/rewrite.py
+	'zones' => [
```
[the remark this answers is not in the document]
Comment (Universal-Omega) [reply FileBackend.php:61]: Maybe not though I can't remember if I set it to go to dbname/ImportDump or just /ImportDump when making that...

File: GlobalSettings.php [hunk 2]
```diff
@@ -287,10 +295,22 @@
 	];
 }
 
+if ( $wi->isExtensionActive( 'score' ) ) {
```
[the remark this answers is not in the document]
Comment (paladox) [reply GlobalSettings.php]: Thanks, done.

File: GlobalSettings.php [hunk 3]
```diff
@@ -287,10 +295,22 @@
...
+	if ( $wmgEnableSwift ) {
+		$wgScoreFileBackend = 'miraheze-swift';
+	}
+}
+
+if ( $wi->isExtensionActive( 'timeline' ) ) {
```
[the remark this answers is not in the document]
Comment (paladox) [reply GlobalSettings.php]: Thanks, done.

File: FileBackend.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (paladox) [reply FileBackend.php:61]: Thanks, done.

**Review by Universal-Omega — APPROVED** [review]
Looks good to me now, provided it actually works, since I only reviewed the code here I can't guarantee anything else.

**Pull request merged**